aiida/cmdline/params/options/config.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida_core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=cyclic-import
+"""
+.. py:module::config
+    :synopsis: Convenience class for configuration file option
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import yaml
+import click_config_file
+
+from .overridable import OverridableOption
+
+
+def yaml_config_file_provider(file_path, cmd_name):  # pylint: disable=unused-argument
+    """Read yaml config file."""
+    try:
+        with open(file_path, 'r') as handle:
+            return yaml.load(handle)
```

Comment:
What will `yaml.load` throw if the content is invalid `yaml`? Should we also catch this and just return empty dict? Maybe add a unit test for this option

Reply:
Actually, what I would prefer is for the exception to simply propagate to the top - I put this try/except in as a hack because the error message was unusable.
I've opened a PR on `click_config_file` to fix this https://github.com/phha/click_config_file/pull/2

Reply:
Fixing the error message surfaced another problem: by default, the option is *always* looking for a config file (by default in a standard directory with a standard name). I.e. if I don't put the try/except here it will always fail.

This is being addressed here (this PR is still to be discussed): https://github.com/phha/click_config_file/pull/4

Since the error messages now are processed correctly, a yaml parser error would just bubble up to the top, so I think it's fine for us for the moment.